src/server/server.rs

```diff
@@ -344,6 +396,33 @@ where
         Ok(())
     }
 
+    pub fn pause(&mut self) -> Result<()> {
+        info!("pausing the grpc server");
+        // Prepare the builder for resume grpc server.
+        // If the builder cannot be created, then pause will be skipped;
+        let builder = Either::Left(self.builder_factory.create_builder(self.env.clone())?);
+        if let Some(Either::Right(server)) = self.builder_or_server.take() {
+            drop(server);
+        }
+        self.health_service
+            .set_serving_status("", ServingStatus::NotServing);
+        self.builder_or_server = Some(builder);
+        Ok(())
+    }
+
+    pub fn resume(&mut self) -> Result<()> {
```

Comment:
It's better to measure the duration in pause/resume as they seem pretty heavy.

Reply:
Got